app/app.go

```diff
@@ -352,6 +359,21 @@ func NewOsmosisApp(
 
 	app.sm.RegisterStoreDecoders()
 
+	// initialize lanes + mempool
+	mevLane, freeLane, defaultLane := CreateLanes(app, txConfig)
+
+	// create the mempool
+	lanedMempool, err := block.NewLanedMempool(
+		app.Logger(),
+		[]block.Lane{mevLane, freeLane, defaultLane},
+	)
```

Comment:
How does `LanedMempool` `Select` transactions from lanes given that `nil` is returned?
https://github.com/skip-mev/block-sdk/blob/31def89bc1fa8e3460a11b53030b6d965da860f4/block/mempool.go#L105-L113

Reply:
Great question! See [here](https://github.com/skip-mev/block-sdk/blob/31def89bc1fa8e3460a11b53030b6d965da860f4/abci/abci.go#L48) for the Prepare / ProcessProposal handlers. Essentially, how blocks are built is that as opposed to calling `app.Mempool.Select()` (which is a no-op), we iterate through `app.LanedMempool.Lanes()`, and call `lane.Select()` on each of the sub-lanes. From that lane we select up to the max-block-space / max-txs allocated for that lane. Does that make sense?